File: Java - Old/app/src/main/java/com/sathwikd/pdftoimage/ThemeUtilsBasic.java
package com.sathwikd.pdftoimage;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.appcompat.app.AppCompatDelegate;

public class ThemeUtilsBasic {

    private static final String SHARED_PREFS = "shared_prefs";
    private static final String THEME_STATUS = "theme_status";

    public static void loadTheme(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
        int themeValue = sharedPreferences.getInt(THEME_STATUS, AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
        AppCompatDelegate.setDefaultNightMode(themeValue);
    }

    public static void saveTheme(Context context, int themeValue) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(THEME_STATUS, themeValue).apply();
    }
}

File: app/src/main/java/com/sathwikd/pdftoimage/ThemeUtilsBasic.kt
package com.sathwikd.pdftoimage

import android.app.Activity
import android.content.Context
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatDelegate
import androidx.core.app.ComponentActivity
import androidx.core.view.WindowCompat

object ThemeUtilsBasic {

    private const val SHARED_PREFS = "shared_prefs_theme"
    private const val THEME_STATUS = "theme_status"

    fun applyThemeAndEdgeToEdge(activity: androidx.activity.ComponentActivity) {
        activity.enableEdgeToEdge()
        loadTheme(activity)
    }

    private fun loadTheme(context: Context) {
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        val themeValue = sharedPreferences.getInt(THEME_STATUS, AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
        AppCompatDelegate.setDefaultNightMode(themeValue)
    }

    fun saveTheme(context: Context, themeValue: Int) {
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        sharedPreferences.edit().putInt(THEME_STATUS, themeValue).apply()
    }
}
